src/main/java/org/broadinstitute/hellbender/utils/read/FlowBasedReadUtils.java

```diff
@@ -26,17 +29,36 @@ public class FlowBasedReadUtils {
     static public class ReadGroupInfo {
         final public String  flowOrder;
         final public int     maxClass;
-
+        final public boolean isFlowPlatform;
         private String  reversedFlowOrder = null;
 
         public ReadGroupInfo(final SAMReadGroupRecord readGroup) {
-
-            Utils.nonNull(readGroup);
-            this.flowOrder = readGroup.getFlowOrder();
-            Utils.nonNull(this.flowOrder);
-
-            String mc = readGroup.getAttribute(FlowBasedRead.MAX_CLASS_READ_GROUP_TAG);
-            this.maxClass = (mc == null) ? FlowBasedRead.MAX_CLASS : Integer.parseInt(mc);
+            if (readGroup.getPlatform()==null){
+                isFlowPlatform = false;
+            } else if (NGSPlatform.fromReadGroupPL(readGroup.getPlatform())==NGSPlatform.UNKNOWN){
+                isFlowPlatform = false;
+            } else {
+                //old Ultima data can have PLATFORM==LS454
+                // we require also FO field to consider ReadGroup to come from the flow
+                boolean tmp = (NGSPlatform.fromReadGroupPL(readGroup.getPlatform()) == NGSPlatform.LS454) ||
+                        (NGSPlatform.fromReadGroupPL(readGroup.getPlatform()) == NGSPlatform.ULTIMA);
+                tmp = tmp & (readGroup!=null);
+                tmp = tmp  & (readGroup.getFlowOrder()!=null);
+                isFlowPlatform = tmp;
+
+                //in addition validate that the ultima data has FO tag, otherwise break
+                if ((NGSPlatform.fromReadGroupPL(readGroup.getPlatform()) == NGSPlatform.ULTIMA) & (!isFlowPlatform)){
```

Comment:
the logic here is a little confusing though I believe it accomplishes what you want. You might consider refactoring this a little so its cleaner but its not critical.

Reply:
Fixed